pkg/operator/controller/controller.go

```diff
@@ -95,10 +101,24 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		ingress = nil
 	}
 
-	caSecret, err := r.ensureRouterCACertificateSecret()
-	if err != nil {
-		errs = append(errs, fmt.Errorf("failed to ensure CA secret: %v", err))
-	} else {
+	caSecretName := cacert.CASecretName(r.Namespace)
+	caSecret := &corev1.Secret{}
+	if err := r.Client.Get(context.TODO(), caSecretName, caSecret); err != nil {
+		if errors.IsNotFound(err) {
+			// Most likely the CA cert controller hasn't created it yet, try again
+			// after a reasonable time.
+			result.RequeueAfter = 5 * time.Second
+			// TODO: This check won't be necessary if all cert stuff is extracted.
+			if ingress != nil {
+				r.recorder.Event(ingress, "Warning", "DefaultWildcardCACertMissing", "The default wildcard CA certificate is missing")
```

Comment:
This feels like more of a status issue: failing condition is true if the CA is missing.

Reply:
I think you're right but I didn't want to expand status with this PR